### PR TITLE
Feature/hee 106 blogpost page

### PR DESCRIPTION
**Pull request by adaMiclos**
Create blogPost document type and blogPost component. 

This implementation doesn't contain blog subscription and blog comment functionality. 

Jira ticket: https://manifesto.atlassian.net/browse/HEE-106

### CODE REVIEW

File: repository-data/development/src/main/resources/hcm-content/content/documents/administration/lks/lksconfiguration.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/content/documents/administration/lks/lksconfiguration:
```
Comment (thojava) [inline repository-data/development/src/main/resources/hcm-content/content/documents/administration/lks/lksconfiguration.yaml:1]: I don't think it is good idea to put the configuration as resource bundle. That is out of the main purpose of resource bundle. Should we find other way ?

File: repository-data/application/src/main/resources/hcm-config/namespaces/hee/blogPage.yaml [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+          /publicationDate:
+            jcr:primaryType: hipposysedit:field
+            hipposysedit:mandatory: false
+            hipposysedit:multiple: false
+            hipposysedit:ordered: false
+            hipposysedit:path: hee:publicationDate
+            hipposysedit:primary: false
+            hipposysedit:type: Date
+            hipposysedit:validators: [required]
+          /category:
```
Comment (thojava) [inline repository-data/application/src/main/resources/hcm-config/namespaces/hee/blogPage.yaml:77]: Should we rename this property to be `categories` as this is for multiple values ?

File: repository-data/development/src/main/resources/hcm-content/content/documents/administration/lks/lksconfiguration.yaml (hunk 1 above)
Comment (adaMiclos) [reply repository-data/development/src/main/resources/hcm-content/content/documents/administration/lks/lksconfiguration.yaml:1]: Sure, please let me know what you have in mind.

File: repository-data/application/src/main/resources/hcm-config/namespaces/hee/blogPage.yaml (hunk 2 above)
Comment (adaMiclos) [reply repository-data/application/src/main/resources/hcm-config/namespaces/hee/blogPage.yaml:77]: Yes, I'll do that.

File: repository-data/development/src/main/resources/hcm-content/content/documents/administration/lks/lksconfiguration.yaml (hunk 1 above)
Comment (thojava) [reply repository-data/development/src/main/resources/hcm-content/content/documents/administration/lks/lksconfiguration.yaml:1]: For this case of storing the blogsOverview's URL, I am thinking
- First we created hstSiteMapItem for the blogsOverview and set its `hst:refId` to a fixed value like `blogs`
- Create the HstLink for blogsOverview page using the fixed refId
```
final HstLink hstLink = request.getRequestContext().getHstLinkCreator().createByRefId("blogs", request.getRequestContext().getResolvedSiteMapItem().getResolvedMount().getMount());
```
- From the HstLink we can have the URL somehow.

Comment (smanivasag) [reply repository-data/development/src/main/resources/hcm-content/content/documents/administration/lks/lksconfiguration.yaml]: @thojava - Updated `uk.nhs.hee.web.components.BlogPostComponent` to generate Blog Overview URL for categories based on `blogs` SiteMapItem RefId and defaults to `/blogs` URL if `blogs` SiteMapItem RefId isn't available in the repository yet. Note that the creation of SiteMapItem with RefId `blogs` should be done as part of blogs overview page build ticket.

File: repository-data/application/src/main/resources/hcm-config/namespaces/hee/blogPage.yaml (hunk 2 above)
Comment (smanivasag) [reply repository-data/application/src/main/resources/hcm-config/namespaces/hee/blogPage.yaml]: @thojava - The field `category` has now been renamed to `categories` 👍

**Comment by smanivasag** [comment]
@thojava - As part of latest commits to this ticket, have refactored `bulletin` with `bulletin-listing` for the clarity of template/component purpose. Hope that makes sense ? [Since, we don't have many Bulletin pages in our env yet, I think we can apply this change on brCloud env manually]

@thojava - Based on my observation so far and if my understanding is correct, I think we may need to perform the following:
- Refactor `/content/documents/administration/valuelists/kls/categories` to (say) `/content/documents/administration/valuelists/kls/bulletinCategories` (as these are Bulletin categories) ?
- Retain generic ListingPageComponent logic in `uk.nhs.hee.web.components.ListingPageComponent` and Extract out the Bulletin specific logics into a separate component class `uk.nhs.hee.web.components.BulletinListingPageComponent` so that `uk.nhs.hee.web.components.ListingPageComponent` could be extended by other `hee:listingPage` document based pages.

Thoughts ?

Fyi - @duncanheron  @MattBrooksManifesto

**Comment by thojava** [comment]
@smanivasag 
> Refactor /content/documents/administration/valuelists/kls/categories to (say) /content/documents/administration/valuelists/kls/bulletinCategories (as these are Bulletin categories) ?
Yes. I would be agree with this.
Same with refactor the ListingPageComponent

**Comment by smanivasag** [comment]
> @smanivasag
> 
> > Refactor /content/documents/administration/valuelists/kls/categories to (say) /content/documents/administration/valuelists/kls/bulletinCategories (as these are Bulletin categories) ?
> > Yes. I would be agree with this.
> > Same with refactor the ListingPageComponent

Thanks @thojava. Are you happy to get this merged to master ? If so, could you approve the PR ? Thanks!

@duncanheron @MattBrooksManifesto  - Would you mind adding a ticket in the backlog with details indicated in the comment https://github.com/Manifesto-Digital/hee-cms-platform/pull/41#issuecomment-824147882 ? Thank you!

**Review by duncanheron — APPROVED** [review]

**Pull request merged**